packages/grafana-data/src/events/EventBus.ts

```diff
@@ -99,24 +99,26 @@ export class EventBusSrv implements EventBus, LegacyEmitter {
  * Wraps EventBus and adds a source to help with identifying if a subscriber should react to the event or not.
  */
 export class EventBusWithSource implements EventBus {
-  source: string;
-  eventBus: EventBus;
-
-  constructor(eventBus: EventBus, source: string) {
-    this.eventBus = eventBus;
-    this.source = source;
-  }
+  constructor(private eventBus: EventBus, private localOnly: boolean) {}
 
   publish<T extends BusEvent>(event: T): void {
     const decoratedEvent = {
       ...event,
-      ...{ payload: { ...event.payload, ...{ source: this.source } } },
+      ...{ payload: { ...event.payload, ...{ source: this } } },
```

Comment:
this makes me a bit nervous for mousemove events -- that is a lot of iterating/copying if the speed is fast.

Reply:
we could do this mutably instead, event.source = this

Reply:
yes, i think we should mutate when possible here. since everything is on one thread, uPlot sometimes has dummy global objects that it simply sets properties on and re-emits.